main.go

```diff
@@ -106,36 +115,61 @@ func main() {
 	flagset.StringVar(&cfg.auth.Authentication.Header.UserFieldName, "auth-header-user-field-name", "x-remote-user", "The name of the field inside a http(2) request header to tell the upstream server about the user's name")
 	flagset.StringVar(&cfg.auth.Authentication.Header.GroupsFieldName, "auth-header-groups-field-name", "x-remote-groups", "The name of the field inside a http(2) request header to tell the upstream server about the user's groups")
 	flagset.StringVar(&cfg.auth.Authentication.Header.GroupSeparator, "auth-header-groups-field-separator", "|", "The separator string used for concatenating multiple group names in a groups header field's value")
+
+	//Authn OIDC flags
+	flagset.StringVar(&cfg.auth.Authentication.OIDC.IssuerURL, "oidc-issuer", "", "Issuer URL")
+	flagset.StringVar(&cfg.auth.Authentication.OIDC.ClientID, "oidc-clientID", "", "Registered clientID at Identity Provider")
+	flagset.StringVar(&cfg.auth.Authentication.OIDC.GroupsClaim, "oidc-groups-claim", "groups", "Identifier of groups in JWT claim, by default set to 'groups'")
+	flagset.StringVar(&cfg.auth.Authentication.OIDC.UsernameClaim, "oidc-username-claim", "email", "Identifier of the user in JWT claim, by default set to 'email'")
+	flagset.StringVar(&cfg.auth.Authentication.OIDC.GroupsPrefix, "oidc-groups-prefix", "", "Delimeter for groups in JWT claim")
+	flagset.StringArrayVar(&cfg.auth.Authentication.OIDC.SupportedSigningAlgs, "oidc-sing-alg", []string{"RS256"}, "Supported signing algorithms, default RS256")
+	flagset.StringVar(&cfg.auth.Authentication.OIDC.CAFile, "oidc-ca-file", "", "Path to OIDC CA certificate file")
+
 	flagset.Parse(os.Args[1:])
+	kcfg := initKubeConfig()
 
 	upstreamURL, err := url.Parse(cfg.upstream)
 	if err != nil {
 		glog.Fatalf("Failed to build parse upstream URL: %v", err)
 	}
 
-	kcfg, err := rest.InClusterConfig()
-	if err != nil {
-		glog.Fatalf("Failed to build Kubernetes rest-config: %v", err)
-	}
-
 	kubeClient, err := kubernetes.NewForConfig(kcfg)
 	if err != nil {
 		glog.Fatalf("Failed to instantiate Kubernetes client: %v", err)
 	}
 
-	if cfg.resourceAttributesFile != "" {
-		b, err := ioutil.ReadFile(cfg.resourceAttributesFile)
+	var authenticator authenticator.Request
+	// If OIDC configuration provided, use oidc authenticator
+	if &cfg.auth.Authentication.OIDC != nil {
```

Comment:
The flags framework will always set `cfg.auth.Authentication.OIDC` to be non-nil. I suggest to add this check instead:
```
if cfg.auth.Authentication.OIDC.IssuerURL != ""
```

Otherwise, by starting the proxy it will panic like this:
```
$ kube-rbac-proxy 
F0829 12:30:28.325892   23957 main.go:146] Failed to instantiate OIDC authenticator: 'oidc-issuer-url' ("") has invalid scheme (""), require 'https'
goroutine 1 [running]:
github.com/brancz/kube-rbac-proxy/vendor/github.com/golang/glog.stacks(0xc420264100, 0xc420066000, 0x95, 0xe9)
	/home/sur/go/src/github.com/brancz/kube-rbac-proxy/vendor/github.com/golang/glog/glog.go:769 +0xcf
...
```